src/NuGet.Core/NuGet.Build.Tasks.Console/ConsoleLoggingQueue.cs

```diff
@@ -0,0 +1,194 @@
+        int IBuildEngine.LineNumberOfTaskNode => 0;
+
+        /// <inheritdoc cref="ILogger.Parameters" />
+        string ILogger.Parameters { get; set; }
+
+        /// <inheritdoc cref="IBuildEngine.ProjectFileOfTaskNode" />
+        string IBuildEngine.ProjectFileOfTaskNode => null;
+
+        /// <summary>
+        /// Gets a <see cref="Microsoft.Build.Utilities.TaskLoggingHelper" /> that can be used to write log messages to the current queue.
+        /// </summary>
+        public TaskLoggingHelper TaskLoggingHelper => _taskLoggingHelperLazy.Value;
+
+        /// <inheritdoc cref="ILogger.Verbosity" />
+        public LoggerVerbosity Verbosity
+        {
+            get => _verbosity;
+            set
+            {
+                _verbosity = value;
+
+                // Determine the minimum verbosity of messages
+                switch (value)
+                {
+                    case LoggerVerbosity.Quiet:
+                    case LoggerVerbosity.Minimal:
+                        _minMessageImportance = MessageImportance.High;
+                        break;
+
+                    case LoggerVerbosity.Normal:
+                        _minMessageImportance = MessageImportance.Normal;
+                        break;
+
+                    case LoggerVerbosity.Detailed:
+                    case LoggerVerbosity.Diagnostic:
+                        _minMessageImportance = MessageImportance.Low;
+                        break;
+
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(value), value, null);
+                }
+            }
+        }
+
+        /// <inheritdoc cref="IBuildEngine.BuildProjectFile" />
+        bool IBuildEngine.BuildProjectFile(string projectFileName, string[] targetNames, IDictionary globalProperties, IDictionary targetOutputs) => throw new NotImplementedException();
+
+        /// <inheritdoc cref="ILogger.Initialize" />
+        void ILogger.Initialize(IEventSource eventSource)
+        {
+            _eventSource = eventSource;
+
+            _eventSource.MessageRaised += OnMessageRaised;
+            _eventSource.WarningRaised += OnWarningRaised;
+            _eventSource.ErrorRaised += OnErrorRaised;
+        }
+
+        /// <inheritdoc cref="IBuildEngine.LogCustomEvent" />
+        void IBuildEngine.LogCustomEvent(CustomBuildEventArgs e) { }
+
+        /// <inheritdoc cref="IBuildEngine.LogErrorEvent" />
+        void IBuildEngine.LogErrorEvent(BuildErrorEventArgs e) => OnErrorRaised(this, e);
+
+        /// <inheritdoc cref="IBuildEngine.LogMessageEvent" />
+        void IBuildEngine.LogMessageEvent(BuildMessageEventArgs e) => OnMessageRaised(this, e);
+
+        /// <inheritdoc cref="IBuildEngine.LogWarningEvent" />
+        void IBuildEngine.LogWarningEvent(BuildWarningEventArgs e) => OnWarningRaised(this, e);
+
+        /// <inheritdoc cref="ILogger.Shutdown" />
+        void ILogger.Shutdown()
+        {
+            if (_eventSource != null)
```

Comment:
The `Initialize` method is called by MSBuild and `eventSource` is never null.  So if Initialize isn't called by MSBuild because of some exception, then `_eventSource` could be null.